app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -3280,7 +3284,16 @@ public void onStartDrag(final PlayQueueItemHolder viewHolder) {
     @Override // own playback listener
     @Nullable
     public MediaSource sourceOf(final PlayQueueItem item, final StreamInfo info) {
-        return (isAudioOnly ? audioResolver : videoResolver).resolve(info);
+        if (audioPlayerSelected()) {
+            return audioResolver.resolve(info);
+        } else {
+            if (isAudioOnly
+                    && !videoResolver.wasLastResolvedVideoAndAudioSeparated().orElse(false)) {
+                return audioResolver.resolve(info);
+            }
+
+            return videoResolver.resolve(info);
```

Comment:
Add a comment here explaining why this is being done (and maybe point to this PR)

Reply:
@TiA4f8R Ping